snippets/javascript/jscalculator.md

```diff
@@ -0,0 +1,48 @@
+# JS based simple calculator
+
+```js
+'use strict'
+
+let operator = '';
+let operation = '';
+let number1;
+let number2;
+
+// accept input from the user and check if it is a number 
+while(isNaN(number1)) {
+ number1 = parseFloat(prompt ('Enter the first number: '));
+}
+  operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
```

Comment:
Take care with the indentation please 🙂.  When skimming over the code, it looks like lines 15-22 are nested in the while of line 12. 

Your code: 
```js
while(isNaN(number1)) {
 number1 = parseFloat(prompt ('Enter the first number: '));
}
  operator = prompt('Enter valid operator ( either +, -, * or / ): ');  

  while(!(operator === '+' || operator === '-' || operator === '*' || operator === '/')){
    operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
  }
  // accept input from the user and check if it is a number 
  while(isNaN(number2)) {  
    number2 = parseFloat(prompt ('Enter the second number: '));
   }
```

How it would look properly formatted:
```js
```js
while(isNaN(number1)) {
 number1 = parseFloat(prompt ('Enter the first number: '));
}
operator = prompt('Enter valid operator ( either +, -, * or / ): ');  

while(!(operator === '+' || operator === '-' || operator === '*' || operator === '/')){
  operator = prompt('Enter valid operator ( either +, -, * or / ): ');  
}
// accept input from the user and check if it is a number 
while(isNaN(number2)) {  
  number2 = parseFloat(prompt ('Enter the second number: '));
}
```
```

Reply:
Line 15 seems to be unnecessary, as it is already being executed on line 17